### PR TITLE
W-13145677: Modularization of Core Components, Deployment, Deployment Model Impl, Extensions Soap Support and Extension Support

### PR DESCRIPTION
**Pull request by mbuchwald**

### CODE REVIEW

File: modules/core-components/src/main/java/module-info.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    requires org.mule.runtime.api;
+    requires org.mule.runtime.core;
+    requires org.mule.runtime.extension.model;
+    requires org.mule.runtime.tracer.api;
+    requires org.mule.runtime.tracer.customization.api;
+
+    requires jakarta.activation;
+    requires java.xml;
+    requires org.apache.log4j;
```
Comment (elrodro83) [inline modules/core-components/src/main/java/module-info.java:23]: not slf4j? is this ok?

File: modules/core-components/src/main/java/module-info.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ */
+module org.mule.runtime.core.components {
+
+    requires org.mule.runtime.api;
+    requires org.mule.runtime.core;
+    requires org.mule.runtime.extension.model;
+    requires org.mule.runtime.tracer.api;
+    requires org.mule.runtime.tracer.customization.api;
+
+    requires jakarta.activation;
```
Comment (elrodro83) [inline modules/core-components/src/main/java/module-info.java:21]: is this for media type handling?

File: modules/core-components/src/main/java/module-info.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    requires java.xml;
```
Comment (elrodro83) [inline modules/core-components/src/main/java/module-info.java:22]: where is this used?

File: modules/deployment-model-impl/src/main/java/module-info.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ */
+/**
+ * Mule Deployment Model Implementation Module
+ *
+ * @moduleGraph
+ * @since 4.6
+ */
+module org.mule.runtime.deployment.model.impl {
+
+    //requires org.mule.runtime.api;
```
Comment (elrodro83) [inline modules/deployment-model-impl/src/main/java/module-info.java:15]: remove

File: modules/deployment/src/main/java/module-info.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    requires org.mule.runtime.artifact.ast.xmlParser;
+    requires org.mule.runtime.artifact.declaration;
+    requires org.mule.runtime.container;
+    requires org.mule.runtime.core;
+    requires org.mule.runtime.deployment.model;
+    requires org.mule.runtime.deployment.model.impl;
+    requires org.mule.runtime.properties.config;
+    requires org.mule.runtime.spring.config;
+
+    requires java.desktop;
```
Comment (elrodro83) [inline modules/deployment/src/main/java/module-info.java:29]: add comment mentioning where/for what this is used

File: modules/extensions-support/src/main/java/module-info.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    requires org.mule.runtime.metadata.support;
+    requires org.mule.runtime.oauth.api;
+    requires org.mule.runtime.policy.api;
+    requires org.mule.runtime.tracer.api;
+    requires org.mule.runtime.tracer.customization.api;
+    requires org.mule.runtime.tracer.internal.impl;
+    requires org.mule.sdk.api;
+    requires org.mule.sdk.compatibility.api;
+
+    requires org.apache.logging.log4j.core;
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/module-info.java:40]: is this using log4j directly?

File: modules/extensions-support/src/main/java/module-info.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    requires org.joda.time;
+    requires com.github.benmanes.caffeine;
+    requires com.google.common;
+    requires spring.core;
+
+    requires java.compiler;
+    requires java.desktop;
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/module-info.java:47]: add comments

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: modules/deployment-model-impl/src/main/java/module-info.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+/**
+ * Mule Deployment Model Implementation Module
+ *
+ * @moduleGraph
+ * @since 4.6
+ */
+module org.mule.runtime.deployment.model.impl {
```
Comment (elrodro83) [inline modules/deployment-model-impl/src/main/java/module-info.java:13]: beware, there is a split package (`org.mule.runtime.deployment.model.internal.domain`) between deployment model jar and its test jar

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: core/src/main/java/module-info.java [hunk 9]
```diff
@@ -444,6 +449,9 @@
       org.mule.runtime.spring.config,
       spring.beans;
 
+  exports org.mule.runtime.core.internal.profiling.context to
```
Comment (elrodro83) [inline core/src/main/java/module-info.java:452]: sort these alphabetically

File: modules/artifact-activation/src/main/java/module-info.java [hunk 10]
```diff
@@ -51,11 +54,19 @@
...
+      org.mule.runtime.deployment.model.impl,
       org.mule.runtime.spring.config,
       org.mule.runtime.log4j,
       org.mule.test.runner;
   exports org.mule.runtime.module.artifact.activation.internal.nativelib to
       org.mule.runtime.deployment.model,
+      org.mule.runtime.deployment.model.impl,
       org.mule.test.runner;
 
+  exports org.mule.runtime.module.artifact.activation.internal.deployable to
```
Comment (elrodro83) [inline modules/artifact-activation/src/main/java/module-info.java:66]: sort these alphabetically

File: modules/artifact/src/main/java/module-info.java [hunk 11]
```diff
@@ -34,11 +34,16 @@
...
+      org.mule.runtime.deployment.model.impl,
       org.mule.test.runner;
   exports org.mule.runtime.module.artifact.internal.util to
       org.mule.runtime.container,
       org.mule.runtime.artifact.activation,
       org.mule.runtime.deployment.model,
+      org.mule.runtime.deployment.model.impl,
       org.mule.test.runner;
 
+  exports org.mule.module.artifact.classloader to
```
Comment (elrodro83) [inline modules/artifact/src/main/java/module-info.java:46]: sort these alphabetically

File: modules/deployment-model-impl/src/main/java/module-info.java [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+        org.mule.runtime.deployment;
+
+    exports org.mule.runtime.module.deployment.impl.internal.artifact to
+        org.mule.runtime.deployment;
+
+    exports org.mule.runtime.module.deployment.impl.internal.application to
+        org.mule.runtime.deployment;
+
+    exports org.mule.runtime.module.deployment.impl.internal.domain to
+        org.mule.runtime.deployment;
```
Comment (elrodro83) [inline modules/deployment-model-impl/src/main/java/module-info.java:45]: sort these alphabetically

File: modules/deployment-model/src/main/java/module-info.java [hunk 13]
```diff
@@ -35,9 +35,18 @@
...
   exports org.mule.runtime.deployment.model.internal.artifact to
-      org.mule.runtime.log4j;
+      org.mule.runtime.log4j,
+      org.mule.runtime.deployment,
+      org.mule.runtime.deployment.model.impl;
+
+  exports org.mule.runtime.deployment.model.internal.policy to
+      org.mule.runtime.deployment.model.impl;
+
+  exports org.mule.runtime.artifact.activation.internal to
```
Comment (elrodro83) [inline modules/deployment-model/src/main/java/module-info.java:49]: this package is not in this module

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: modules/deployment/src/main/java/module-info.java [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+    requires spring.core;
+
+    exports org.mule.runtime.module.deployment.api;
+
```
Comment (elrodro83) [inline modules/deployment/src/main/java/module-info.java:33]: add `    exports org.mule.runtime.module.deployment.internal.processor to
        org.mule.runtime.launcher;`

File: modules/deployment-model-impl/src/main/java/module-info.java [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+    exports org.mule.runtime.module.deployment.impl.internal.util to
+        org.mule.runtime.deployment;
+
+    exports org.mule.runtime.module.deployment.impl.internal.artifact to
+        org.mule.runtime.deployment;
+
+    exports org.mule.runtime.module.deployment.impl.internal.application to
+        org.mule.runtime.deployment;
+
+    exports org.mule.runtime.module.deployment.impl.internal.domain to
```
Comment (elrodro83) [inline modules/deployment-model-impl/src/main/java/module-info.java:44]: add `    exports org.mule.runtime.module.deployment.impl.internal to
        org.mule.runtime.launcher;`

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: modules/core-components/src/main/java/module-info.java (hunk 1 above)
Comment (mbuchwald) [reply modules/core-components/src/main/java/module-info.java:23]: Log4j is used [in logger](https://github.com/mulesoft/mule/blob/master/modules/core-components/src/main/java/org/mule/runtime/core/internal/processor/LoggerMessageProcessor.java#L31)

File: modules/core-components/src/main/java/module-info.java (hunk 2 above)
Comment (mbuchwald) [reply modules/core-components/src/main/java/module-info.java:21]: Yep, https://github.com/mulesoft/mule/blob/master/modules/core-components/src/main/java/org/mule/runtime/core/internal/processor/simple/ParseTemplateProcessor.java#L42

File: modules/core-components/src/main/java/module-info.java (hunk 3 above)
Comment (mbuchwald) [reply modules/core-components/src/main/java/module-info.java:22]: [QName](https://github.com/mulesoft/mule/blob/master/modules/core-components/src/main/java/org/mule/runtime/core/internal/routing/forkjoin/CollectListForkJoinStrategyFactory.java#L29)

File: modules/deployment/src/main/java/module-info.java (hunk 14 above)
Comment (elrodro83) [reply modules/deployment/src/main/java/module-info.java:33]: add `exports org.mule.runtime.module.deployment.internal to
        org.mule.runtime.launcher;`

File: modules/artifact/src/main/java/module-info.java [hunk 16]
```diff
@@ -34,11 +34,16 @@
       org.mule.runtime.artifact.activation,
       org.mule.runtime.extensions.support,
       org.mule.runtime.deployment.model,
+      org.mule.runtime.deployment.model.impl,
```
Comment (elrodro83) [inline modules/artifact/src/main/java/module-info.java:37]: add `org.mule.runtime.launcher` too

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: modules/extensions-support/src/main/java/module-info.java (hunk 6 above)
Comment (mbuchwald) [reply modules/extensions-support/src/main/java/module-info.java:40]: There was an incorrect import [here](https://github.com/mulesoft/mule/blob/master/modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/streaming/PagingResultTransformer.java#L9), I removed it and everything is working properly.

**Pull request closed without merging**